### ISSUE TITLE
Narrowing types using a metaclass can cause false positives

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#10423]
The `type(t) is not M` check should not narrow the type of `t`, since a subclass can have the metaclass:

```py
from typing import Type

class M(type):
    pass

class C: pass

class D(C, metaclass=M): pass

def f(t: Type[C]) -> None:
    if type(t) is not M:
        return
    reveal_type(t)  # <nothing>

f(D)
```

I'd expect that the revealed type to be `Type[C]`, not `<nothing>`. The latter can cause false positives.

This is a regression from mypy 0.812.
